src/OrchardCore.Modules/OrchardCore.Redis.Azure/AzureRedisTokenProvider.cs

```diff
@@ -0,0 +1,47 @@
+using Azure.Core;
+using Microsoft.Extensions.Options;
+using OrchardCore.Azure.Core;
+
+namespace OrchardCore.Redis.Azure;
+
+public sealed class AzureRedisTokenProvider : ITokenProvider
+{
+    private readonly IOptionsMonitor<AzureOptions> _options;
+    private readonly RedisOptions _redisOptions;
+
+    public AzureRedisTokenProvider(
+        IOptionsMonitor<AzureOptions> options,
+        IOptions<RedisOptions> redisOptions)
+    {
+        _options = options;
+        _redisOptions = redisOptions.Value;
+    }
+
+    public async Task<TokenResult> GetTokenAsync()
+    {
+        var redisOptions = _options.Get(_redisOptions.CredentialName ?? AzureOptions.DefaultName);
+
+        var scope = redisOptions.GetProperty<string>("Scope");
```

Comment:
This should be in the Azure module, and be configured in the credentials.

Reply:
I am not follow you here